### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by MenendezON**
This request is about the :
- Create an interface.
- Make sure that my Person class inherits from Nameable
- Make sure that this class has a method correct_name implemented. It should simply return the name attribute.
- Make sure that it inherits from Nameable.
- In the constructor assign a nameable object from params to an instance variable.
- Implement the correct_name method that returns the result of the correct_name method of the @nameable.

### CODE REVIEW

**Comment by KDlamini** [comment]
## Status: Invalid 🔴 

### Hi @MenendezON 🙋‍♀️,  

While you made a great effort in this project, I can't review your code.

## Invalid Code Review Request
 
You have submitted a project that has a README that is not professional. Here are key points to consider when documenting your README.

- The License suggestion should be removed. 
image

- Please make sure that the information in the README is relevant to the project. Surely the project title cannot be `postgresSQL` and the `key features` are for the decode message project.


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_Invalid Code Review Request **does not count** into the code reviews limit._

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Decorator < Nameable
+  def initialize(nameable)
```
Comment (MahmoudBakr23) [inline decorator.rb:4]: - [ ] Please make sure that you specify the relationship between the Decorator class and the Nameable class to ensure the correctness of the inheritance.

attr_accessor :nameable

File: trimmer_decorator [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class TrimmerDecorator < Decorator
+  def correct_name
+    super[0, 10]
```
Comment (MahmoudBakr23) [inline trimmer_decorator]: - [ ] Please make sure to write the correct method for the correct name in the trimmer class.
The name has a maximum of 10 characters. If it's longer it should trim the word.

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @MenendezON 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done on making the classes and the custom methods so far
- Descriptive pr and readme file
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ademibowale — APPROVED** [review]
Hi @MenendezON  , 


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️
-  Check if Person class inherits from Nameable class.  ✔️
- Check for a Namable class with a method 'correct_name' rasing an error.  s ✔️
- Descriptive commit messages ✔️ 
- Good use of Gitflow ✔️






- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @Adebowale ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**